### ISSUE TITLE
internet-proxy saturation due to COA concourse docker registry access

### ISSUE THREAD

**Issue by poblin-orange** [orange-cloudfoundry/paas-templates#463]
### Expected behavior
As a platform operator, i need a stable COA concourse environement

### Observed behavior
on 41.0.0, under high load (many  launched in //), the micro-depls/internet-proxy has cpu saturation.
```
resource script '/opt/resource/check []' failed: exit status 1

stderr:
failed to ping registry: 2 error(s) occurred:

* ping https: Get https://registry-1.docker.io/v2/: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)
* ping http: Get http://registry-1.docker.io/v2/: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)

```

relates to #363

### Affected release

Reproduced on version 41.0.0

### Workaround

reboot the vm

**Comment by poblin-orange** [comment]
ongoing fix for 41.0.1

**Comment by gberche-orange** [comment]
Additional symptoms

Bosh reports the squid process as failing

```
Deployment 'internet-proxy'

Instance                                                    Process State  AZ  IPs  
system-internet-relay/6a764944-d362-4eaa-b5ce-a2a6dc471bb9  failing        z1  192.168.116.80  
```

Monit attempts to restart the squid process in loop

```
system-internet-relay/6a764944-d362-4eaa-b5ce-a2a6dc471bb9:~# cat /var/vcap/monit/monit.log

[UTC Jun 25 16:00:54] info     : 'squid' trying to restart
[UTC Jun 25 16:00:54] info     : 'squid' start: /var/vcap/jobs/squid/bin/squid
[UTC Jun 25 16:01:24] error    : 'squid' failed to start
[UTC Jun 25 16:01:34] error    : 'squid' process is not running
[UTC Jun 25 16:01:34] info     : 'squid' trying to restart
```

Squid PID is present but points to a dead process

```
ls -al /var/vcap/sys/run/squid/squid.pid; cat /var/vcap/sys/run/squid/squid.pid; ps -p $(cat  /var/vcap/sys/run/squid/squid.pid)
-rw-r--r-- 1 vcap vcap 6 Jun 26 07:15 /var/vcap/sys/run/squid/squid.pid
28939
  PID TTY          TIME CMD
```

However squid is running and handling some request before being killed by monit which waits for the wrong pid

```
system-internet-relay/6a764944-d362-4eaa-b5ce-a2a6dc471bb9:~# ps -ef | grep squid
vcap     15955     1  0 Jun25 ?        00:00:00 /var/vcap/packages/squid/sbin/squid -f /var/vcap/jobs/squid/config/squid.conf
vcap     15957 15955  0 Jun25 ?        00:03:35 (squid-1) -f /var/vcap/jobs/squid/config/squid.conf
vcap     15959 15957  0 Jun25 ?        00:00:10 (logfile-daemon) /var/vcap/sys/log/squid/access.log
root     16193 15368  0 Jun25 pts/0    00:00:00 tail -f /var/vcap/sys/log/squid/access.log
root     29174 15368  0 07:17 pts/0    00:00:00 grep --color=auto squid
```

Potentially related squid fix/improvement     https://github.com/squid-cache/squid/blob/master/ChangeLog

>    Changes to squid-4.0.20 (01 Jun 2017):
>    - Make PID file check/creation atomic to avoid associated race conditions

**Comment by poblin-orange** [comment]
concourse http proxy finally disabled. merged for v41.0.1
